crypto/rand/rand_lib.c

```diff
@@ -241,12 +243,10 @@ size_t rand_drbg_get_additional_data(unsigned char **pout, size_t max_len)
 #endif
 
 #ifdef OPENSSL_SYS_UNIX
-    if (tsc == 0 && clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
-        RAND_POOL_add(pool, (unsigned char *)&ts, sizeof(ts), 0);
-    if (clock_gettime(CLOCK_REALTIME, &ts) == 0)
-        RAND_POOL_add(pool, (unsigned char *)&ts, sizeof(ts), 0);
+    if (gettimeofday(&tv, NULL) == 0)
+        RAND_POOL_add(pool, (unsigned char *)&tv, sizeof(tv), 0);
 #elif defined(OPENSSL_SYS_WIN32)
-    if (tsc == 0 && QueryPerformanceCounter(&pc) != 0)
+    if (QueryPerformanceCounter(&pc) != 0)
```

Comment:
Yeah, that's a good approach.  Frankly, I don't quite understand why `tsc` should stop us from add *yet another few bits*, but then, entropy isn't my field of expertise...

Reply:
The performance counter (and CLOCK_MONOTONIC) have the tsc as source. If we can read it ourself, I don't see why we should read it a second time with a possible high overhead.